editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/CamelGlobalConfigEditor.java

```diff
@@ -730,7 +714,7 @@ public CamelBean addNewGlobalBeanElement(CamelFile cf, Node newXMLNode) {
 	private void configureCamelModelElement(CamelFile cf, Element newXMLNode, AbstractCamelModelElement cme, final String eipName) {
 		cme.setXmlNode(newXMLNode);
 		IProject project = cf.getResource().getProject();
-		final CamelModel camelModel = CamelCatalogCacheManager.getInstance().getCamelModelForProject(project);
+		final CamelModel camelModel = CamelCatalogCacheManager.getInstance().getCamelModelForProject(project, new NullProgressMonitor());
```

Comment:
The whole reason for this change was to remove the deprecation warning. I did not change this class otherwise and also did not plan to rework it.

Reply:
I'm wondering if it is not better to let the deprecated code in this case.
The goal of the deprecation is to ensure that all callers are using a ProgressMonitor because it can take a long time. If we put a NullProgressMonitor, there will no report to users of this long-running operations and in code we won't see it easily that this can be improved at this place. The deprecation is helping to spot that we have an action to take to improve usability, that we have a real usability issue so the warning is justified.